mono/metadata/unity-liveness.c

```diff
@@ -1,100 +1,150 @@
 #include <config.h>
 #include <glib.h>
-#include <mono/metadata/object.h>
-#include <mono/metadata/object-internals.h>
-#include <mono/metadata/metadata.h>
-#include <mono/metadata/tabledefs.h>
 #include <mono/metadata/class-internals.h>
 #include <mono/metadata/domain-internals.h>
+#include <mono/metadata/metadata.h>
+#include <mono/metadata/object-internals.h>
+#include <mono/metadata/object.h>
+#include <mono/metadata/tabledefs.h>
 #include <mono/utils/mono-error.h>
 
 typedef struct _LivenessState LivenessState;
 
-typedef struct _GPtrArray custom_growable_array;
-#define array_at_index(array,index) (array)->pdata[(index)]
+#define k_block_size (8 * 1024)
+#define k_array_elements_per_block ((k_block_size - 3 * sizeof (void*)) / sizeof (gpointer))
```

Comment:
Nit, maybe not this magic `3` comes from the number of fields in `custom_array_block` I assume?

Reply:
yes. I wanted the array to be 8K large, but with the buffer embedded. I can add a comment about it